internal/feather.go

```diff
@@ -315,7 +323,56 @@ func ColumnMapToRecord(inputs map[string]any) (arrow.Record, error) {
 			return nil, errors.Wrapf(err, "failed to set values for feature '%s'", field.Name)
 		}
 	}
-	return recordBuilder.NewRecord(), nil
+
+	res := recordBuilder.NewRecord()
+
+	// Need to convert __ts__ to exactly pa.timestamp("us", "UTC")
+	// until the UploadFeatures gRPC endpoint handles precisions
+	// other than "us".
+	tsIndex := -1
+	var tsField arrow.Field
+	for idx, field := range res.Schema().Fields() {
+		if field.Name == "__ts__" {
+			tsIndex = idx
+			tsField = field
+			break
+		}
+	}
+	if tsIndex != -1 {
```

Comment:
this is possibly made redundant with Raphael's change in the backend to cast colujmns accordingly

Reply:
removed